src/components/form/form.test.js

```diff
@@ -0,0 +1,221 @@
+  return (
+    <Form
+      onSubmit={() => console.log("submit")}
+      saveButton={
+        <Button buttonType="primary" type="submit">
+          Save
+        </Button>
+      }
+      {...props}
+    >
+      <Textbox label="Textbox1" />
+      <Textbox label="Textbox2" />
+      <Textbox label="Textbox3" />
+    </Form>
+  );
+};
+
+context("Tests for Form component", () => {
+  describe("check props for Form component", () => {
+    it.each([
+      ["left", "start"],
+      ["right", "end"],
+    ])(
+      "should check buttonAlignment set as %s for Form component",
+      (buttonAlignment, webkitAlign) => {
+        CypressMountWithProviders(
+          <FormComponent buttonAlignment={buttonAlignment} />
+        );
+        formFooterComponent().should(
+          "have.css",
+          "-webkit-box-pack",
+          webkitAlign
+        );
+      }
+    );
+
+    it("should check children for Form component", () => {
+      CypressMountWithProviders(
+        <Form>
+          <Textbox label="Textbox" />
+        </Form>
+      );
+      getElement("input").should("be.visible");
+    });
+
+    it.each([3, 11, 9])(
+      "should check errorCount as %s for Form component",
+      (errorCount) => {
+        CypressMountWithProviders(<FormComponent errorCount={errorCount} />);
+        getElement("errors")
+          .should("have.css", "color", "rgb(162, 44, 59)")
+          .and("have.text", `${errorCount} errors`)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      [0, 287],
+      [3, 335],
+      [5, 367],
+      [7, 407],
+    ])(
+      "should check fieldSpacing as %s for Form component",
+      (fieldSpacing, formHeight) => {
+        CypressMountWithProviders(
+          <FormComponent fieldSpacing={fieldSpacing} />
+        );
+        FormPreview().then(($el) => {
+          useJQueryCssValueAndAssert($el, "height", formHeight);
+        });
+      }
+    );
+
+    it("should check leftSideButtons for Form component", () => {
+      CypressMountWithProviders(
+        <FormComponent
+          leftSideButtons={
+            <Button onClick={() => console.log("cancel")} ml={2}>
+              Cancel
+            </Button>
+          }
+        />
+      );
+      formButtonComponent()
+        .eq(0)
+        .should("have.text", "Cancel")
+        .and("be.visible");
+    });
+
+    it.each([true, false])(
+      "should check noValidate set as %s for Form component",
+      (noValidate) => {
+        CypressMountWithProviders(<FormComponent noValidate={noValidate} />);
+
+        if (noValidate === true) {
+          FormPreview().should("have.attr", "noValidate");
+        } else {
+          FormPreview().should("not.have.attr", "noValidate");
+        }
+      }
+    );
+
+    describe("check events for form component", () => {
+      let callback;
+      beforeEach(() => {
+        callback = cy.stub();
+      });
+      it("should call onSubmit callback when a click event is triggered", () => {
```

Comment:
if we have only 1 test for `onSubmit` why we need to create a separate `describe` block?
Even if we've created it why we need to put all other tests here?